keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -0,0 +1,181 @@
+import { Dialog } from '@headlessui/react';
+import { useState } from 'react';
+import Select, { components, SingleValue, MenuListProps} from 'react-select';
+import { TextInput, Button, Subtitle } from '@tremor/react';
+import { AuthenticationType } from "utils/authenticationType";
+import { User } from "./models";
+import { getApiURL } from "utils/apiUrl";
+
+interface RoleOption {
+    value: string;
+    label: string | JSX.Element;
+    tooltip?: string;
+    isDisabled?: boolean;
+  }
+
+interface AddUserModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  authType: string;
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>;
+  accessToken: string;
+}
+
+export default function AddUserModal({ isOpen, onClose, authType, setUsers, accessToken }: AddUserModalProps) {
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [selectedRole, setSelectedRole] = useState<SingleValue<RoleOption>>(null);
+  const [password, setPassword] = useState('');
+  const [hoveredOption, setHoveredOption] = useState<string | null>(null);
+  const [addUserError, setAddUserError] = useState('');
+
+  let roleOptions: RoleOption[] = [
```

Comment:
+ move this outside of the function so it doesn't re-render all the time
```suggestion
  const roleOptions: RoleOption[] = [
```